controllers/zync_controller.go

```diff
@@ -72,76 +73,27 @@ func (r *ZyncReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		instance.Spec,
 	)
 
-	// Calculate rollout triggers
-	triggers, err := r.TriggersFromSecretDefs(ctx,
-		gen.ZyncSecretDefinition(),
-	)
+	// Shared resources
+	resources := gen.Resources()
+
+	// Api resources
+	var api_resources []basereconciler.Resource
+	api_resources, err = r.NewDeploymentWorkloadWithTraffic(ctx, instance, r.GetScheme(), &gen.API, &gen.API)
```

Comment:
Whenever possible the short variable declaration form can be used:

```suggestion
	api_resources, err := r.NewDeploymentWorkloadWithTraffic(ctx, instance, r.GetScheme(), &gen.API, &gen.API)
```

This helps make the code more readable and compact.